libs/platform/src/lib/components/form/radio-group/radio-group.component.ts

```diff
@@ -0,0 +1,282 @@
+import {
+    AfterContentInit,
+    Component,
+    Output,
+    ContentChildren,
+    EventEmitter,
+    forwardRef,
+    QueryList,
+    Input,
+    ChangeDetectorRef,
+    Optional,
+    Self,
+    OnChanges
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { RadioButtonComponent, stateType } from '@fundamental-ngx/core';
+import { SelectItem } from '../data-model';
+import { BaseInput } from '../base.input';
+
+// Increasing integer for generating unique ids for radio components.
+let nextUniqueId = 0;
+
+/** Change event object emitted by RadioButtonComponent and RadioGroupComponent. */
+export class RadioButtonChange {
+    constructor(
+        // The RadioButtonComponent that emits the change event.
+        public source: RadioButtonComponent,
+        // The value of the RadioButtonComponent.
+        public value: any
+    ) {}
+}
+
+@Component({
+    selector: 'fdp-radio-group',
+    templateUrl: './radio-group.component.html'
+})
+export class RadioGroupComponent extends BaseInput implements OnChanges, AfterContentInit {
+    /**
+     * values of radio buttons. type will be of SelectItem or String
+     */
+    @Input()
+    list: Array<SelectItem | string>;
+
+    /**
+     * if all radio buttons are compact size.
+     */
+    @Input()
+    compact: boolean = false;
+
+    /**
+     * Type of Radio buttons.
+     */
+    @Input()
+    state: stateType = 'default';
+
+    /**
+     * To Dispaly Radio buttons in a line or not
+     */
+    @Input()
+    get isInline(): boolean {
+        return this._isInline;
+    }
+
+    set isInline(value: boolean) {
+        this._isInline = coerceBooleanProperty(value);
+    }
+
+    /**
+     * When the radio is required show None value to let user know to select something
+     */
+    @Input()
+    hasNoValue: boolean = false;
+
+    @Input()
+    noValueLabel: string = 'None';
+
+    /** Name of the radio button group. All radio buttons inside this group will use this name. */
+    @Input()
+    get name(): string {
+        return this._name;
+    }
+    set name(value: string) {
+        if (this.name !== value) {
+            this._name = value;
+            this._updateRadioButtonNames();
```

Comment:
@fkolar ,as content is not initialised. I am not able to update names for content Radio buttons. please suggest